mypyc/irbuild/specialize.py

```diff
@@ -302,3 +302,20 @@ def translate_isinstance(builder: IRBuilder, expr: CallExpr, callee: RefExpr) ->
         if irs is not None:
             return builder.builder.isinstance_helper(builder.accept(expr.args[0]), irs, expr.line)
     return None
+
+
+@specialize_function('builtins.min')
+def faster_min(builder: IRBuilder, expr: CallExpr, callee: RefExpr) -> None:
+    if (len(expr.args) > 0
```

Comment:
The return value should be `Optional[Value]`. The the call translation functions in `irbuild/expression.py` can get the correct result and stop further trying.